legend-engine-language-pure-grammar/src/main/java/org/finos/legend/engine/language/pure/grammar/from/domain/DomainParser.java

```diff
@@ -110,4 +115,14 @@ public ValueSpecification parseCombinedExpression(String code, ParseTreeWalkerSo
         DomainParseTreeWalker walker = new DomainParseTreeWalker(walkerSourceInformation, parserContext, allowPropertyBracketExpression);
         return walker.combinedExpression(((DomainParserGrammar) sectionParserInfo.parser).combinedExpression(), "line", typeParametersNames, lambdaContext, "", true, false);
     }
+
+    public ValueSpecification parseInstanceRightSide(String code, ParseTreeWalkerSourceInformation instanceWalkerSourceInformation, PureGrammarParserContext parserContext, boolean allowPropertyBracketExpression)
+    {
+        List<String> typeParametersNames = new ArrayList<>();
+        DomainParseTreeWalker.LambdaContext lambdaContext = new DomainParseTreeWalker.LambdaContext("");
+        ParseTreeWalkerSourceInformation walkerSourceInformation = new ParseTreeWalkerSourceInformation.Builder(instanceWalkerSourceInformation).build();
+        SourceCodeParserInfo sectionParserInfo = this.getParserInfo(code, null, walkerSourceInformation, false);
+        DomainParseTreeWalker walker = new DomainParseTreeWalker(walkerSourceInformation, parserContext, allowPropertyBracketExpression);
+        return walker.combinedExpression(((DomainParserGrammar) sectionParserInfo.parser).combinedExpression(), "line", typeParametersNames, lambdaContext, "", true, false);
```

Comment:
Should this be combinedExpression or instanceRightSide?

Reply:
should be combined expression only as DomainParseTreeWalker doesn't have support for instanceRightSide, so extending the support for combined expression only as that can be extended to instanceRightSide as it covers only a subset of the possibilities covered in combinedExpression

Reply:
changed to processrRghtSide as discussed